core/src/main/scala/variant/Horde.scala

```diff
@@ -67,6 +67,9 @@ case object Horde
   override def opponentHasInsufficientMaterial(position: Position): Boolean =
     hasInsufficientMaterial(position.board, !position.color) || hordeClosedPosition(position)
 
+  override def playerHasInsufficientMaterial(position: Position): Boolean =
+    throw Exception("`playerHasInsufficientMaterial` is not implemented yet for Horde")
```

Comment:
We can't have `Variant` offer a `playerHasInsufficientMaterial(position: Position): Boolean` function that sometimes throws an exception.

This would either create bugs, or force us to code defensively by catching exceptions. It's not how we do things 'round here.

If `Variant.playerHasInsufficientMaterial` can't always return a `Boolean`, then it must return an `Option[Boolean]` to let the type system know what's up, and force callers to act accordingly.

Reply:
@ornicar Updated to do this. Before merging, do you think it'd be good to add tests in each variant for `playerHasInsufficientMaterial`? 

- In most variants, this would just look like making sure the result is the same as `opponentHasInsufficientMaterial` (with colours swapped and preserving turn, to test that turn doesn't give a different result). Maybe these two checks could be grouped in a function?
- In Horde, a test would just make sure the function returns None.
- In Antichess, there are already tests from this PR.

Reply:
> Variant.playerHasInsufficientMaterial could return Boolean with false mean we don't know.

I disagree, I think there's a useful distinction between "no" and "I don't know", that Option[Boolean] provides here. The caller decides whether to treat them differently or not.

Reply:
I believe for these `*HasInsufficientMaterial` we want to guarantee two things:
- No false positive
- They have to be consistent between each other

Reply:
> We don't need to test it, but are we 100% sure that this affirmation is true?
> 
> > For all variants except Antichess and Horde, considering turn isn't needed
> 
> If you are certain that it's true for standard chess especially, then I think the PR is good to go.

I'm not sure this is 100 % true either